tests/tel2puml/find_unique_graphs/test_json_data_converter.py

```diff
@@ -298,3 +299,111 @@ def test_process_header(
+                    "spans": [
+                        {
+                            "trace_id": "trace001",
+                            "span_id": "span001",
+                            "parent_span_id": None,
+                        }
+                    ],
+                    "spans_not_in_list": {
+                        "trace_id": "trace001",
+                        "span_id": "span001",
+                        "parent_span_id": None,
+                    },
+                    "empty_spans": [],
+                    "more_than_one_span": [
+                        {"span1": "value1"},
+                        {"span2": "value2"},
+                    ],
+                    "nested_span": {"spans": [{"span3": "value3"}]},
+                }
+            ],
+            "more_than_one_scope_spans": [
+                {"spans": [{"span_id": "001"}]},
+                {"item2": "value2"},
+            ],
+            "nested_scope_span": {
+                "scope_span": [
+                    {"nested_span": {"spans": [{"span5": "value5"}]}},
+                ]
+            },
+        }
+
+        json_config: JSONDataSourceConfig = mock_yaml_config_dict[
+            "data_sources"
+        ]["json"]
+
+        # Test single span
+        spans = process_spans(json_config, sample_data)
+        assert spans == [
+            {
+                "trace_id": "trace001",
+                "span_id": "span001",
+                "parent_span_id": None,
+            }
+        ]
+
+        # Test multiple spans
+        json_config["span_mapping"]["spans"]["key_paths"] = [
+            "scope_spans::more_than_one_span"
+        ]
+        spans = process_spans(json_config, sample_data)
+        assert spans == [
+            {"span1": "value1"},
+            {"span2": "value2"},
+        ]
+
+        # Test nested spans
+        json_config["span_mapping"]["spans"]["key_paths"] = [
+            "scope_spans::nested_span:spans"
+        ]
+        spans = process_spans(json_config, sample_data)
+        assert spans == [{"span3": "value3"}]
+
+        # Test nested scope spans and nested spans
+        json_config["span_mapping"]["spans"]["key_paths"] = [
+            "nested_scope_span:scope_span::nested_span:spans"
+        ]
+        spans = process_spans(json_config, sample_data)
+        assert spans == [{"span5": "value5"}]
+
+        # Test spans not within a list
+        json_config["span_mapping"]["spans"]["key_paths"] = [
+            "scope_spans::spans_not_in_list"
+        ]
+        with pytest.raises(TypeError):
+            process_spans(json_config, sample_data)
+
+        # Test incorrect use of ::
+        json_config["span_mapping"]["spans"]["key_paths"] = [
+            "scope_spans::nested_span::spans"
+        ]
+        with pytest.raises(TypeError):
+            process_spans(json_config, sample_data)
+
+        # Test an empty list of spans
+        json_config["span_mapping"]["spans"]["key_paths"] = [
```

Comment:
Not sure if we want to error on this as there could always be a chance of an empty list - then if we've done several other files we'll lose all that previous work. Might be worth using a logger for this.

Reply:
Updated to use a logger